harmony/http.py

```diff
@@ -251,7 +258,34 @@ def _log_download_performance(logger, url, duration_ms, file_size):
     logger.info('timing.download.end', extra=extra_fields)
 
 
-def download(config, url: str, access_token: str, data, destination_file, user_agent=None):
+def _preprocess_download_kwargs(download_kwargs: dict, logger):
+    """Preprocess the download kwargs.
+
+    Parameters
+    ----------
+    download_kwargs: dict
+        kwargs to be passed to the download agent
+        E.g. stream=True
+    logger : logging.Logger
+        The logger to use.
+    NOTE: By default, streaming request is used to download the file,
+          and the chunksize is set to be 16MB based on the experiment with a large file of 1.8Gb
+          for optimized speed and memory consumption.
+    """
+    download_kwargs.setdefault('stream', True)
+    download_kwargs.setdefault('chunk_size', 1024*1024*16)
+    stream = download_kwargs['stream'] if 'stream' in download_kwargs else False
```

Comment:
Because you set defaults on the previous two lines `if 'stream' in download_kwargs` will always be true.  Similar for the next line and `chunk_size`.  Also `setdefault` returns the value of the property setting the default, so these four lines are equivalent to:

```
stream = download_kwargs.setdefault('stream', True)
chunk_size = download_kwargs.setdefault('chunk_size', 1024*1024*16)
```